content/docs/principles/proudly-invented-elsewhere.md

```diff
@@ -6,3 +6,9 @@ description: Software development is not new. That means somebody might have alr
 # Proudly Invented Elsewhere
 
 Software development is not new. That means somebody might have already faced a similar problem and has solved it. Use libraries whenever you can. Even though a library does 70-80% of your requirement, consider using it.
+
+We prefer using established, contributed libraries over writing custom code. This approach is more cost-effective and requires fewer programming skills for maintenance. These projects often have established best practices and allow us to standardize our processes. To implement this, we download and configure relevant libraries. We choose to use these libraries when their functions closely align with our requirements, anticipated function changes can be accommodated through configuration, and other users should be able to make configuration changes.
+
+There are cases where we should avoid using contributed libraries. If a lighter-weight approach can achieve the same functionality, if using other tools provides more configurability and less lock-in, if the complexity of setup and configuration outweighs the benefits, or if directly modifying the database might be more efficient, we might opt for custom code.
```

Comment:
> if using other tools provides more configurability and less lock-in

I don't understand how this fits in the list. Everything else justifies writing custom code but this point is a benefit in using contributed libraries.

> or if directly modifying the database might be more efficient

Can we find a more generic example? This is not inaccurate but all the other points in the list are quite generic and this is too specific. This breaks the flow of reading.

Reply:
@hussainweb: Accepted and made changes accordingly.